app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -183,27 +184,26 @@ public void handleTaskUpdate(Integer... vals) {
 
     @Override
     public void handleTaskCompletion(AppInstallStatus result) {
+        if (CommCareApplication._().isConsumerApp()) {
+            dismissProgressDialog();
+        }
+
         if (result == AppInstallStatus.UpdateStaged) {
             uiController.unappliedUpdateAvailableUiState();
             if (proceedAutomatically) {
                 launchUpdateInstallTask();
-                return;
```

Comment:
realized that these 3 return statements were unnecessary, and removing them allowed me not to duplicate the call to `unregisterTask()` (which also occurs immediately below)